packages/browser/src/integrations/graphqlClient.ts

```diff
@@ -0,0 +1,139 @@
+import { SENTRY_XHR_DATA_KEY } from '@sentry-internal/browser-utils';
+import { getBodyString } from '@sentry-internal/replay';
+import {
+  SEMANTIC_ATTRIBUTE_HTTP_REQUEST_METHOD,
+  SEMANTIC_ATTRIBUTE_SENTRY_OP,
+  SEMANTIC_ATTRIBUTE_URL_FULL,
+  defineIntegration,
+  spanToJSON,
+} from '@sentry/core';
+import type { Client, HandlerDataFetch, HandlerDataXhr, IntegrationFn } from '@sentry/types';
+import { getGraphQLRequestPayload, isString, parseGraphQLQuery, stringMatchesSomePattern } from '@sentry/utils';
+
+interface GraphQLClientOptions {
+  endpoints: Array<string | RegExp>;
+}
+
+// Standard graphql request shape: https://graphql.org/learn/serving-over-http/#post-request-and-body
+interface GraphQLRequestPayload {
+  query: string;
+  operationName?: string;
+  variables?: Record<string, unknown>;
+  extensions?: Record<string, unknown>;
+}
+
+const INTEGRATION_NAME = 'GraphQLClient';
+
+const _graphqlClientIntegration = ((options: GraphQLClientOptions) => {
+  return {
+    name: INTEGRATION_NAME,
+    setup(client) {
+      _updateSpanWithGraphQLData(client, options);
+      _updateBreadcrumbWithGraphQLData(client, options);
+    },
+  };
+}) satisfies IntegrationFn;
+
+function _updateSpanWithGraphQLData(client: Client, options: GraphQLClientOptions): void {
+  client.on('beforeOutgoingRequestSpan', (span, handlerData) => {
+    const spanJSON = spanToJSON(span);
+
+    const spanAttributes = spanJSON.data || {};
+    const spanOp = spanAttributes[SEMANTIC_ATTRIBUTE_SENTRY_OP];
+
+    const isHttpClientSpan = spanOp === 'http.client';
+
+    if (isHttpClientSpan) {
```

Comment:
l: I would invert this and just early-return, makes this a bit easier to read IMHO :) so:

```js
if (!isHttpClientSpan) {
  return;
}
// rest of logic here...
```